### DISCUSSION TITLE
IModalService - Inject vs CascadingParameter

### DISCUSSION THREAD

**Discussion by Jimmys20** [Blazored/Modal#527] [Q&A] [answered]
Hello, I just noticed that in my code I've been getting an `IModalService` instance sometimes using `[Inject]` and other times using `[CascadingParameter]`. `CascadingParameter` is how it's done in the documentation. Still I haven't noticed any issues using [Inject] so far. Are there any side effects to injecting `IModalService`?

**Comment by chrissainty** [comment, marked as answer]
No, there isn't as long as you have the `<CascadingBlazoredModal>` component added. Under the hood, the `CascadingParameter` is the `IModalService` from DI. The reason we do this is to make sure the service and the `<CascadingBlazoredModal>` component have been setup correctly. If a developer forget to register the service with the DI container, or forgets the `<CascadingBlazoredModal>` component, there will be an error straight away.

**Reply by Jimmys20** [reply]
Thanks!
